contracts/lib/OnChainIdentity.sol

```diff
@@ -0,0 +1,277 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.16;
+pragma abicoder v2;
+
+import "../interfaces/IState.sol";
+import "../lib/SmtLib.sol";
+import "../lib/Poseidon.sol";
+import "../lib/GenesisUtils.sol";
+
+// /**
+//  * @dev Contract managing onchain identity
+//  */
+library OnChainIdentity {
+    using SmtLib for SmtLib.Data;
+
+    uint256 public constant IDENTITY_MAX_SMT_DEPTH = 40;
+    /**
+     * @dev Identity
+     * Id
+     * Identity latest state
+     * Is old state genesis flag
+     * State contract
+     * Mapping of roots by state
+     * Trees
+     * Last tree roots
+     */
+    struct Identity {
+        uint256 id;
+        uint256 latestState;
+        bool isOldStateGenesis;
+        IState stateContract;
+        mapping(uint256 => Roots) rootsByState;
+        Trees trees;
+        Roots lastTreeRoots;
+        // This empty reserved space is put in place to allow future versions
+        // of the SMT library to add new Data struct fields without shifting down
+        // storage of upgradable contracts that use this struct as a state variable
+        // (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps)
+        uint256[43] __gap;
+    }
+
+    /**
+     * @dev SMT addresses
+     */
+    struct Trees {
+        SmtLib.Data claimsTree;
+        SmtLib.Data revocationsTree;
+        SmtLib.Data rootsTree;
+    }
+
+     /**
+     * @dev set of roots
+     */
+    struct Roots {
+        uint256 claimsRoot;
+        uint256 revocationsRoot;
+        uint256 rootsRoot;
+    }
+
+    function initialize(
+        Identity storage self,
+        address _stateContractAddr,
+        address _identityAddr,
+        uint256 depth
+    ) external {
+        require(depth <= IDENTITY_MAX_SMT_DEPTH,  "SMT depth shouldn't be more than 40" );
```

Comment:
Change the message to "SMT depth is greater than max allowed depth" or something similar to avoid hardcoding exact data in the messages?

Reply:
@volodymyr-basiuk please do this fix

Reply:
@OBrezhniev  fixed this one as well as other comments: 
https://github.com/iden3/contracts/pull/158/files